src/sagas/backend.ts

```diff
@@ -136,6 +143,79 @@ function* backendSaga(): SagaIterator {
       yield call(showWarningMessage, "Couldn't reach our servers. Are you online?")
     }
   })
+
+  yield takeEvery(actionTypes.FETCH_GRADING_OVERVIEWS, function*() {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }))
+    const gradingOverviews = yield call(getGradingOverviews, tokens)
+    if (gradingOverviews) {
+      yield put(actions.updateGradingOverviews(gradingOverviews))
+    }
+  })
+
+  yield takeEvery(actionTypes.FETCH_GRADING, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }))
+    const id = (action as actionTypes.IAction).payload
+    const grading = yield call(getGrading, id, tokens)
+    if (grading) {
+      yield put(actions.updateGrading(id, grading))
+    }
+  })
+
+  yield takeEvery(actionTypes.SUBMIT_GRADING, function*(action) {
+    const {
+      submissionId,
+      questionId,
+      grade,
+      comment,
+      adjustment
+    } = (action as actionTypes.IAction).payload
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }))
+    const resp = yield postGrading(submissionId, questionId, grade, comment, adjustment, tokens)
+    if (resp !== null && resp.ok) {
+      yield call(showSuccessMessage, 'Saved!', 1000)
+      // Now, update the grade for the question in the Grading in the store
+      const grading: Grading = yield select((state: IState) =>
+        state.session.gradings.get(submissionId)
+      )
+      const newGrading = grading.slice().map((gradingQuestion: GradingQuestion) => {
+        if (gradingQuestion.question.id === questionId) {
+          gradingQuestion.grade = {
+            adjustment,
+            comment,
+            grade
+          }
+        }
+        return gradingQuestion
+      })
+      yield put(actions.updateGrading(submissionId, newGrading))
+    } else if (resp !== null) {
+      let errorMessage: string
+      switch (resp.status) {
+        case 400:
+          errorMessage = 'Invalid or missing parameter(s) or submission and/or question not found'
+          break
+        case 401:
+          errorMessage = 'Got 403 response. Only staff can save gradings.'
```

Comment:
Typo here (should be '403'). Also, can you help me change [this line](https://github.com/source-academy/cadet-frontend/blob/c61fd6009f480bccc4ed06470a364e5ae9c98b4b/src/sagas/backend.ts#L130) to `case 401`?

Anyways I think this case clause can be omitted. Anyone using the site properly would be a staff by the point they reach here, so it doesn't make sense to remind them that only a staff can save gradings. This should fall in the default clause of 'oops we don't know what happened, here's the error code, go open an issue please'.